IReflection methods connected with xml serialization rename

whut opened the pull request:
Hi,

Here is a patch with renames of methods on IReflection connected to xml serialization to be more xml centric, this way I can be sure that my custom implementation of IReflection will not break other parts of RSB in future, other than xml serialization:)
- GetType is now GetTypeFromXmlNamespace
- GetAssemblyQualifiedNameWithoutVersion is now GetNamespaceForXml
- GetNamespaceForXml is now GetNamespacePrefixForXml

In my project I created custom implementations of IReflection.GetAssemblyQualifiedNameWithoutVersion and IReflection.GetType, to change the way RSB serialize messages to xml. Both methods are used only in XmlMessageSerializer, first to create xml namespace for given type, second to recreate type from such xml namespace. I needed my custom implementation because, giving example, we have Service project with messages, and Client project, that instead of referencing to Service.dll, compile those message into its own assembly. The problem was that serialization on Client sets xml namespaces to something like "Service.Feature.MyMessage, Client", but on Service during deserialization expected type would be "Service.Feature.MyMessage, **Service**", and other way around: message send from Service will have namespace pointing to Service.dll, but Client have those messages in Client.dll.

CoreyKaylor commented:
Any chance we can send a pull request for items that apply to the request only? The precondition change to psake script won't run either of the Compile tasks if the precondition is false. It really isn't necessary to add the batch file. If it makes it easier on you create a simple branch for your needs. Although all the whitespace changes are with good intention it adds noise to the request at hand.

whut committed:
Changed IReflection methods connected with xml serialization to better names.

Better means that there are more intention revealing, and not so generic, like GetType renamed to GetTypeFromXmlNamespace.
This way I can safely interfere which way IReflection creates xml namespaces and so on without risk of breaking future code, that is not connected to xml serialization

whut commented:
Here is the most important part: renames of three methods in IReflection. Thanks! (old commits are in branch private_xml_old)

Sorry about bug in Test task, whoops! It could be fixed by adding adding to Release task: "-depends Compile,Test", but yes, that not the most important part of this pull request.
